src/amalgam/gen/avx512f.c

```diff
@@ -5557,3 +5553,103 @@ void xnn_s32_vmulc_ukernel__avx512f_u32(
     xnn_store_tail_s32(output, vy, batch >> XNN_LOG2_SIZEOF_INT32_T);
   }
 }
+
+void xnn_s32_vor_ukernel__avx512f_u32(
+    size_t batch,
+    const int32_t* input_a,
+    const int32_t* input_b,
+    int32_t* output,
+    const union xnn_s32_default_params params[restrict XNN_MIN_ELEMENTS(1)])
+{
+  assert(batch != 0);
+  assert(batch % sizeof(int32_t) == 0);
+  assert(input_b != NULL);
+  assert(input_a != NULL);
+  assert(output != NULL);
+  assert(xnn_simd_size_s32 == 16);
+
+  for (; batch >= 32 * sizeof(int32_t); batch -= 32 * sizeof(int32_t)) {
+    xnn_simd_s32_t vin1_0 = xnn_loadu_s32(input_a);
+    xnn_simd_s32_t vin1_1 = xnn_loadu_s32(input_a + 1 * xnn_simd_size_s32);
+    input_a += 32;
+
+    xnn_simd_s32_t vin2_0 = xnn_loadu_s32(input_b);
+    xnn_simd_s32_t vin2_1 = (xnn_loadu_s32(input_b + 1 * xnn_simd_size_s32));
+    input_b += 32;
+
+    xnn_simd_s32_t vy_0 = xnn_or_s32(vin1_0, vin2_0);
+    xnn_simd_s32_t vy_1 = xnn_or_s32(vin1_1, vin2_1);
+
+    xnn_storeu_s32(output, vy_0);
+    xnn_storeu_s32(output + 1 * xnn_simd_size_s32, vy_1);
+    output += 32;
+  }
+  for (; batch >= xnn_simd_bytes_s32; batch -= xnn_simd_bytes_s32) {
```

Comment:
you use batch -= 32 * sizeof(int32_t)) { for the main loop.  which is the correct amount for avx512
should this one be batch -= 16 * sizeof(int32_t)) {

Reply:
@fbarchard  vin1_0, vin1_1 two set of inputs are processed here, like loop unrolling of 2 so its 32

Reply:
vin1 is from input_a and is 16 ints
vin2 is from input_b and is 16 ints
should the loop be doing:
for (; batch >= 16 * sizeof(int32_t); batch -= 16 * sizeof(int32_t)) {